src/helpers/modeRegistry.ts

```diff
@@ -31,6 +33,24 @@ class ModeRegistry {
         });
     }
 
+    static modesRegisterFromDir(dirPath: string): void {
+        const modes: any[] = [];
+
+        const files = fs.readdirSync(dirPath);
+        files.forEach((f) => {
+            const filePath = path.join(dirPath, f);
+            if (fs.statSync(filePath).isDirectory()) {
+                const modeFilePath = path.join(filePath, `${f}.ts`);
+                if (fs.existsSync(modeFilePath)) {
+                    const modeClass = require(modeFilePath).default;
+                    const modeObject = new modeClass();
```

Comment:
It would be good to validate the `modeObject` before pushing it to the `modes` array. This can help to ensure that only valid modes are registered. [medium]